src/shogun/evaluation/SigmoidCalibration.cpp

```diff
@@ -0,0 +1,158 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are
+ * those
+ * of the authors and should not be interpreted as representing official
+ * policies,
+ * either expressed or implied, of the Shogun Development Team.
+ */
+
+#include <shogun/evaluation/SigmoidCalibration.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/Statistics.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+using namespace shogun;
+
+CSigmoidCalibration::CSigmoidCalibration() : CCalibration()
+{
+	init();
+}
+
+CSigmoidCalibration::~CSigmoidCalibration()
+{
+	SG_FREE(m_sigmoid_parameters);
+}
+
+void CSigmoidCalibration::init()
+{
+	m_sigmoid_parameters = SG_MALLOC(CStatistics::SigmoidParamters, 1);
+}
+
+bool CSigmoidCalibration::fit_binary(CBinaryLabels* predictions, CBinaryLabels* targets)
+{
+	SG_FREE(m_sigmoid_parameters);
+
+	m_sigmoid_parameters = SG_MALLOC(CStatistics::SigmoidParamters, 1);
+
+	m_sigmoid_parameters[0] = CStatistics::fit_sigmoid(predictions->get_values(), targets->get_labels());
+
+	return true;
+}
+
+CBinaryLabels*
+CSigmoidCalibration::calibrate_binary(CBinaryLabels* predictions)
+{
+	auto params = m_sigmoid_parameters[0];
+	
+	/** Convert predictions to probabilties. */
+	auto values = calibrate_values(predictions->get_values(), params);
+	
+	CBinaryLabels* calibrated_predictions = new CBinaryLabels(values);
+
+	return calibrated_predictions;
+}
+
+bool CSigmoidCalibration::fit_multiclass(CMulticlassLabels* predictions, CMulticlassLabels* targets)
+{
+	index_t num_classes =
+	    predictions->get_num_classes();
+
+	SG_FREE(m_sigmoid_parameters);
```

Comment:
as said i would rather use something more high level than this

Reply:
Understood